src/executor.rs

```diff
@@ -91,12 +99,21 @@ impl Executor {
 
     pub fn run(&self, task: Task) {
         let result = self.execute_task(task);
-
         self.finalize_task(result)
     }
 
-    pub fn run_tasks(&mut self) {
+    pub fn run_tasks(&mut self) -> Result<(), FangError> {
         loop {
+            if let Some(ref shared_state) = self.shared_state {
+                let shared_state = shared_state.read()?;
+                match *shared_state {
```

Comment:
this `match` can be replaced with `if let`

Reply:
Fixed, for some reason I remember clippy complaining about it, but not anymore